recordlinkage/fusion.py

```diff
@@ -0,0 +1,1221 @@
+        elif tie_break == 'trust_b' or tie_break == 'last':
+            tie_break_fun = choose_last
+        elif tie_break == 'min':
+            tie_break_fun = choose_min
+        elif tie_break == 'max':
+            tie_break_fun = choose_max
+        elif tie_break == 'shortest':
+            tie_break_fun = choose_shortest_tie_break
+        elif tie_break == 'longest':
+            tie_break_fun = choose_longest_tie_break
+        elif tie_break == 'null' or tie_break == 'nullify':
+            tie_break_fun = nullify
+        else:
+            raise ValueError('Invalid tie_break strategy: {}. Must be one of '
+                             'random, trust_a, trust_b, first, last, min, max, '
+                             'shortest, longest, or nullify.'.format(tie_break))
+    else:
+        raise ValueError('tie_break must be a string or a function.')
+    return tie_break_fun
+
+
+class SkipNull(object):
+    """
+    This object is used like a function decorator for ignoring missing
+    values when applying a transformation function. An object is
+    used in place of a function because local functions cannot
+    be pickled, causing errors during multiprocessing.
+    """
+
+    def __init__(self, f):
+        self.f = f
+
+    def __call__(self, x):
+        if pd.isnull(x):
+            return x
+        else:
+            return self.f(x)
+
+
+@add_metaclass(ABCMeta)
+class FuseCore():
+    def __init__(self):
+        """
+        ``FuseCore`` and its subclasses are initialized without data. The initialized
+        object is populated by metadata describing a series of data resolutions,
+        which are executed when ``.fuse()`` is called.
+        """
+        self.index = None
+        self.predictions = None
+        self.df_a = None
+        self.df_b = None
+        self.suffix_a = None
+        self.suffix_b = None
+        self.resolution_queue = []
+        self._bases_taken = []
+        self._names_taken = []
+        self._sep = ''
+        self._index_level_0 = None
+        self._index_level_1 = None
+
+    def no_gossiping(self, values_a, values_b, name=None):
+        """
+        Handles data conflicts by keeping values agree upon by both data sources,
+        and returning np.nan for conflicting or missing values.
+        
+        Parameters
+        ----------
+        values_a : str/list
+            Column name(s) from df_a to be resolved.
+        values_b : str/list
+            Column name(s) from df_b to be resolved.
+        name : str
+            The name of the resulting resolved column.
+
+        Returns
+        -------
+        None
+            Queues a conflict resolution job for later computation, to be completed when ``fuse`` method is called.
+        """
+        self.resolve(no_gossip, values_a, values_b, name=name, remove_na_vals=False, description='no_gossiping')
+
+    def roll_the_dice(self, values_a, values_b, name=None, remove_na_vals=True):
```

Comment:
The method names are funny to me. I should have named this 'random'. Is it common to name algorithms like these in such a way is academic literature about data fusion?

Reply:
You're right on both counts — the method names are from the academic data fusion literature, but they're also a little funny. Jillian and I have been using the academic names, but I'm happy to change them if that's your preference. As I see it, the pros and cons of the current naming scheme are:

Pros: 
* Consistent with the academic literature.
* Distinguishes between conflict resolution strategies (e.g. `roll_the_dice`) and the conflict resolution functions that implement those strategies (e.g. `choose_random`).

Cons:
* The names are odd and require the user to learn what they mean, which increases the barrier to entry.

If you'd like them changed, let me know.

Reply:
Clear! I prefer the academic literature on data fusion. I do have to read more about data fusion (I am using record linkage techniques for population estimation most of the time).